Source/ACE.Entity/AceObject.cs

```diff
@@ -1816,24 +1816,48 @@ public object Clone(uint guid)
         {
             AceObject ret = (AceObject)Clone();
             ret.AceObjectId = guid;
+            // We are cloning a new AceObject with a new AceObjectID - need to set this to false. Og II
+            ret.HasEverBeenSavedToDatabase = false;
+            ret.IsDirty = true;
+
+            ret.PaletteOverrides.ForEach(c => c.AceObjectId = guid);
+            ret.TextureOverrides.ForEach(c => c.AceObjectId = guid);
+            ret.AnimationOverrides.ForEach(c => c.AceObjectId = guid);
+            ret.IntProperties.ForEach(c => c.AceObjectId = guid);
+            ret.Int64Properties.ForEach(c => c.AceObjectId = guid);
+            ret.DoubleProperties.ForEach(c => c.AceObjectId = guid);
+            ret.BoolProperties.ForEach(c => c.AceObjectId = guid);
+            ret.DataIdProperties.ForEach(c => c.AceObjectId = guid);
+            ret.InstanceIdProperties.ForEach(c => c.AceObjectId = guid);
+            ret.StringProperties.ForEach(c => c.AceObjectId = guid);    
+            ret.GeneratorLinks.ForEach(c => c.AceObjectId = guid);
+            
+            // We're not seemingly changing the guids for the Dictionaries below from Clone(), is this bad?
```

Comment:
The dictionaries separate the DB layer data and the in-game data, so the guid doesn't need to be adjusted (it will be crated/adjusted on save).  This is an okay note, but the comments below it are just clutter.